src/javascript/Edit/Edit.jsx

```diff
@@ -39,13 +46,31 @@ export const EditCmp = ({
                 sections,
                 values
             },
-            editCallback: nodeUuid => {
+            editCallback: (node, mutateNode) => {
                 if (values[Constants.systemFields.OVERRIDE_SUBMIT_CALLBACK]) {
                     values[Constants.systemFields.OVERRIDE_SUBMIT_CALLBACK]();
                 } else {
                     const envEditCallback = envEditCallbacks[contentEditorConfigContext.env];
                     if (envEditCallback) {
-                        envEditCallback(nodeUuid, contentEditorConfigContext);
+                        envEditCallback(node.uuid, contentEditorConfigContext);
+                    }
+                }
+
+                if (mutateNode.rename) {
+                    if (storedLocation && storedLocation.location) {
+                        const newNameStartPosition = mutateNode.rename.lastIndexOf('/') + 1;
+                        const newName = mutateNode.rename.substring(newNameStartPosition, mutateNode.rename.length);
+                        if (openPaths) {
+                            dispatch(replaceOpenedPath(openPaths.map(openPath => openPath.replace(node.name, newName))));
```

Comment:
Dangerous, do not rename other element of the path that should not be rename,
Replace the entire path, instead of the name to avoid that.